adodb-lib.inc.php

```diff
@@ -416,14 +416,15 @@ function _adodb_getcount(&$zthis, $sql,$inputarr=false,$secs2cache=0)
 			} else
 				$rewritesql = "SELECT COUNT(*) FROM (".$rewritesql.")";
 
-		} else if (strncmp($zthis->databaseType,'postgres',8) == 0
-			|| strncmp($zthis->databaseType,'mysql',5) == 0
-			|| strncmp($zthis->databaseType,'mssql',5) == 0
-		) {
-			$rewritesql = "SELECT COUNT(*) FROM ($rewritesql) _ADODB_ALIAS_";
-		} else {
-			$rewritesql = "SELECT COUNT(*) FROM ($rewritesql)";
-		}
+        } else if (strncmp($zthis->databaseType,'postgres',8) == 0
+            || strncmp($zthis->databaseType,'mysql',5) == 0
+	    || strncmp($zthis->databaseType,'mssql',5) == 0
+            || strncmp($zthis->dsnType,'sqlsrv',5) == 0
```

Comment:
@mnewnham if my understanding is correct, this would cover the case of pdo_sqlsrv driver, but what about pdo_mssql ?

Reply:
I do not have a test case but it ought to, because 'mssql' is in the conditional.

Reply:
@dregad , sorry, I keep forgetting about mssql/freetds under unix. I have re-updated the code